gcPauses -> jvm.gc.pause, memory -> jvm.memory.used

BoykoAlex opened the pull request:
- Metric name changes: `gcPauses` -> `jvm.gc.pause`, `memory` -> `jvm.memory.used`. Are those metric names different for other apps?
- Constants for metric names and NPE-safe string equals checks
- `Measurements` value as `Double` rather than `Long`

@vudayani-vmw please have a look at this PR and provide feedback.

vudayani-vmw commented on headless-services/spring-boot-language-server/src/main/java/org/springframework/ide/vscode/boot/java/livehover/v2/SpringProcessConnectorService.java:
```diff
@@ -232,7 +236,7 @@ private void scheduleRefresh(ProgressTask progressTask, SpringProcessParams spri
 			try {
 				progressTask.progressEvent(progressMessage);
-				if(endpoint.equals("metrics") && metricName.equals("memory")) {
+				if(METRICS.equals(endpoint) && MEMORY.equals(metricName)) {
```
The `metricName` passed from the API call here is ("memory", "gcPauses"). This is to identify the type of metric and not the actual metricName used for the actuator call. (Probably I should have named it `metricType`. But I wanted to keep the [metricQuery](https://github.com/spring-projects/sts4/blob/b2af04b6cd32b9a7b44af5676ffe10da5d5004ca/vscode-extensions/vscode-spring-boot/lib/api.d.ts#L81) more generic to be used for other metric calls.

The [getMemoryMetrics](https://github.com/spring-projects/sts4/blob/b2af04b6cd32b9a7b44af5676ffe10da5d5004ca/headless-services/spring-boot-language-server/src/main/java/org/springframework/ide/vscode/boot/java/livehover/v2/SpringProcessLiveDataExtractorOverJMX.java#L189) function is where the metric names ("jvm.memory.used, jvm.memory.max") are passed.

Also, with the current implementation, the [retrieveLiveMemoryMetricsData](https://github.com/spring-projects/sts4/blob/b2af04b6cd32b9a7b44af5676ffe10da5d5004ca/headless-services/spring-boot-language-server/src/main/java/org/springframework/ide/vscode/boot/java/livehover/v2/SpringProcessLiveDataExtractorOverHttp.java#L139) has the logic to fetch the heap and non-heap memory metrics together.

vudayani-vmw commented on headless-services/spring-boot-language-server/src/main/java/org/springframework/ide/vscode/boot/java/livehover/v2/SpringProcessCommandHandler.java:
```diff
@@ -301,11 +301,11 @@ private CompletableFuture<Object> handleLiveMetricsProcessRequest(ExecuteCommand
 					return CompletableFuture.completedFuture(data.getGcPausesMetrics());
 				}
-				case "memory": {
+				case SpringProcessConnectorService.MEMORY: {
```
The `metricName` passed here are ("gcPauses", "heapMemory", "nonHeapMemory"). This is used to fetch the region-specific metric data separately. 
(This is the result of my bad naming convention again ;-) )

@BoykoAlex I would like to discuss and get your feedback on how to improve this.

BoykoAlex commented:
@vudayani-vmw I switched it back to the names you had. I noticed that in[ ](https://github.com/spring-projects/sts4/blob/b2af04b6cd32b9a7b44af5676ffe10da5d5004ca/headless-services/spring-boot-language-server/src/main/java/org/springframework/ide/vscode/boot/java/livehover/v2/SpringProcessLiveDataExtractorOverJMX.java#L189) you're switching the actual metric name. GC Pauses was missing the proper metric name.
Also fixed a few things around logging 404s - don't think we'd want to log 404's as it might be that metric or any other piece of live data hasn't been exposed by the app. Also fixed the case of `{ownerId}` in the URL which was attempted to be subbed by variable value by the REST Template. Furthermore, double encoding should be fixed as well.

BoykoAlex commented:
Closed with a41166383353374d7d422b9496e99c8c425c399f
